app/scripts/controllers/permissions/selectors.js

```diff
@@ -44,14 +44,41 @@ export const getPermittedAccountsByOrigin = createSelector(
   },
 );
 
+/**
+ * Get the authorized CAIP-25 scopes for each subject, keyed by origin.
+ * The values of the returned map are immutable values from the
+ * PermissionController state.
+ *
+ * @returns {Map<string, Caip25Authorization>} The current origin:authorization map.
+ */
+export const getAuthorizedScopesByOrigin = createDeepEqualSelector(
```

Comment:
Given the discussion above, are we okay with using `createSelector` here? @FrederikBolding 

``` suggestion
export const getAuthorizedScopesByOrigin = createSelector(
```

Reply:
Changed back in [825188e](https://github.com/MetaMask/metamask-extension/pull/27782/commits/825188efe7f9dcc69b6f3bf25b1a575ca580e2af).